app/models/foreman_openscap/arf_report.rb

```diff
@@ -1,112 +1,118 @@
-        arf_report = ArfReport.where(:asset_id => asset.id, :policy_id => policy.id,
-                                     :date => params[:date], :digest => digest).first_or_create!
-        if arf_report.arf_report_raw.nil?
-          ArfReportRaw.where(:arf_report_id => arf_report.id, :size => arf_bzip_size, :bzip_data => arf_bzip).create!
-        end
-      end
+    def self.report_status_column
+      "status"
+    end
+
+    def status=(st)
+      s = case st
+          when Integer, Fixnum
+            st
+          when Hash
+            ArfReportStatusCalculator.new(:counters => st).calculate
+          else
+            fail Foreman::Exception(N_('Unsupported report status format'))
+          end
+      write_attribute(:status, s)
     end
 
-    def self.search_by_comply_with(_key, _operator, policy_name)
-      search_by_policy_results policy_name, &:comply
+    delegate :status, :status_of, :to => :calculator
+    delegate(*METRIC, :to => :calculator)
+
+    def calculator
+      ArfReportStatusCalculator.new(:bit_field => read_attribute(self.class.report_status_column))
     end
 
-    def self.search_by_not_comply_with(_key, _operator, policy_name)
-      search_by_policy_results policy_name, &:incomply
+    def passed
+      status_of "passed"
     end
 
-    def self.search_by_inconclusive_with(_key, _operator, policy_name)
-      search_by_policy_results policy_name, &:inconclusive
+    def failed
+      status_of "failed"
     end
 
-    def self.search_by_policy_results(policy_name, &selection)
-      cond = sanitize_sql_for_conditions('foreman_openscap_policies.name' => policy_name)
-      { :conditions => ForemanOpenscap::ArfReport.arel_table[:id].in(
-        ForemanOpenscap::ArfReport.select(ForemanOpenscap::ArfReport.arel_table[:id])
-            .latest.instance_eval(&selection).joins(:policy).where(cond).ast
-        ).to_sql
-      }
+    def othered
+      status_of "othered"
     end
 
-    def self.search_by_last_for(key, operator, by)
-      by.gsub!(/[^[:alnum:]]/, '')
-      case by.downcase
-      when 'host'
-        { :conditions => 'foreman_openscap_arf_reports.id IN (
-              SELECT MAX(id) FROM foreman_openscap_arf_reports sub
-              WHERE sub.asset_id = foreman_openscap_arf_reports.asset_id)' }
-      when 'policy'
-        { :conditions => 'foreman_openscap_arf_reports.id IN (
-              SELECT MAX(id) FROM foreman_openscap_arf_reports sub
-              WHERE sub.policy_id = foreman_openscap_arf_reports.policy_id)' }
-      else
-        fail "Cannot search last by #{by}"
+    def rules_count
+      status.values.sum
+    end
+
+    def self.create_arf(asset, params)
+      # fail if policy does not exist.
+      policy = Policy.find(params[:policy_id])
+      ArfReport.transaction do
+        # TODO:RAILS-4.0: This should become arf_report = ArfReport.find_or_create_by! ...
+        arf_report = ArfReport.create!(:host_id => asset.id,
```

Comment:
`:asset_id => asset.id` should remain or no relationship between asset and report will be created